### PR TITLE
Hidden fixes

### PR DESCRIPTION
**Pull request by duncanmeech**
#### Overview

Can specify construct when focusing blocks. Fixes issues with hidden block selection in EGF templates.

#### Requirements

- [ ] Adds a test (for features + fixes)
- [ ] Docs updated (for features + fixes)

### CODE REVIEW

File: src/actions/focus.js [hunk 1]
```diff
@@ -77,14 +77,17 @@ export const focusBlocks = blockIds => (dispatch, getState) => {
...
       //const constructId = !!construct ? construct.id : (construct !== null ? firstBlockId : undefined);
-    const constructId = construct ? construct.id : null;
-    if (constructId !== focusedConstructId || constructId === firstBlockId) {
+      cid = construct ? construct.id : null;
+    }
+    if (cid !== focusedConstructId || cid === firstBlockId) {
       dispatch({
         type: ActionTypes.FOCUS_CONSTRUCT,
-        constructId,
+        cid,
```
Comment (maxbates) [inline src/actions/focus.js]: this needs to be constructId

File: src/containers/graphics/views/layout.js [hunk 2]
```diff
@@ -725,7 +725,7 @@ export default class Layout {
     let rowIndex = 0;
 
     // display only non hidden blocks
-    const components = this.showHidden ? ct.components : ct.components.filter(blockId => !this.blockIsHidden(blockId));
+    const components = this.showHidden ? [...ct.components] : ct.components.filter(blockId => !this.blockIsHidden(blockId));
```
Comment (maxbates) [inline src/containers/graphics/views/layout.js]: unnecessary unless you want to slice

Comment (duncanmeech) [inline src/containers/graphics/views/layout.js]: Look below the components array is modified for backbone constructs.

**Comment by maxbates** [comment]
#1654

**Pull request closed without merging**